sympy/polys/rootoftools.py

```diff
@@ -394,6 +394,20 @@ def free_symbols(self):
         # symbols
         return set()
 
+    def _eval_power(self, other):
+        if not isinstance(other, (Integer, int)):
+            return None
+        d = degree(self.expr)
```

Comment:
```suggestion
        d = self.poly.degree()
```

Reply:
There are more code below you can just use API of `poly` directly. `degree` internally converts expr back to Poly, and `expr` converts saved attribute `Poly` to `Expr`, so it's better not to round trip